x-pack/plugins/lens/public/datatable_visualization/components/table_basic.test.tsx

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { mountWithIntl } from '@kbn/test/jest';
+import { EuiDataGrid } from '@elastic/eui';
+import { IAggType, IFieldFormat } from 'src/plugins/data/public';
+import { EmptyPlaceholder } from '../../shared_components';
+import { LensIconChartDatatable } from '../../assets/chart_datatable';
+import { DatatableComponent } from './table_basic';
+import { LensMultiTable } from '../../types';
+import { DatatableProps } from '../expression';
+
+function sampleArgs() {
+  const indexPatternId = 'indexPatternId';
+  const data: LensMultiTable = {
+    type: 'lens_multitable',
+    tables: {
+      l1: {
+        type: 'datatable',
+        columns: [
+          {
+            id: 'a',
+            name: 'a',
+            meta: {
+              type: 'string',
+              source: 'esaggs',
+              field: 'a',
+              sourceParams: { type: 'terms', indexPatternId },
+            },
+          },
+          {
+            id: 'b',
+            name: 'b',
+            meta: {
+              type: 'date',
+              field: 'b',
+              source: 'esaggs',
+              sourceParams: {
+                type: 'date_histogram',
+                indexPatternId,
+              },
+            },
+          },
+          {
+            id: 'c',
+            name: 'c',
+            meta: {
+              type: 'number',
+              source: 'esaggs',
+              field: 'c',
+              sourceParams: { indexPatternId, type: 'count' },
+            },
+          },
+        ],
+        rows: [{ a: 'shoes', b: 1588024800000, c: 3 }],
+      },
+    },
+  };
+
+  const args: DatatableProps['args'] = {
+    title: 'My fanci metric chart',
+    columns: {
+      columnIds: ['a', 'b', 'c'],
+      sortBy: '',
+      sortDirection: 'none',
+      type: 'lens_datatable_columns',
+    },
+  };
+
+  return { data, args };
+}
+
+function copyData(data: LensMultiTable): LensMultiTable {
+  return JSON.parse(JSON.stringify(data));
+}
+
+describe('DatatableComponent', () => {
+  let onDispatchEvent: jest.Mock;
+
+  beforeEach(() => {
+    onDispatchEvent = jest.fn();
+  });
+
+  test('it renders the title and value', () => {
+    const { data, args } = sampleArgs();
+
+    expect(
+      shallow(
+        <DatatableComponent
+          data={data}
+          args={args}
+          formatFactory={(x) => x as IFieldFormat}
+          dispatchEvent={onDispatchEvent}
+          getType={jest.fn()}
+          renderMode="edit"
+        />
+      )
+    ).toMatchSnapshot();
```

Comment:
Not a big fan of snapshot tests when the snapshots are so hard to understand. The snapshot file here is 500 lines long: how are we supposed to check for errors?

Reply:
I think this test was here before we had other ones.
I used to spot issues for the `renderMode` flag was set to `edit`, which renders many more feature than when in `display` mode in dashboard.